src/Tribe/Importer/File_Importer_Venues.php

```diff
@@ -36,30 +35,191 @@ protected function create_post( array $record ) {
+	 * @param array $record An event record from the import.
+	 *
+	 * @return array $venue The array of venue data for creation/update.
+	 */
 	private function build_venue_array( $venue_id, array $record ) {
-		$show_map_setting = tribe( 'events-aggregator.settings' )->default_map( 'csv' );
-		$venue_address    = trim( $this->get_value_by_key( $record, 'venue_address' ) . ' ' . $this->get_value_by_key( $record, 'venue_address2' ) );
-		$venue            = array(
-			'Venue'         => $this->get_value_by_key( $record, 'venue_name' ),
-			'Description'   => $this->get_value_by_key( $record, 'venue_description' ),
-			'Address'       => $venue_address,
-			'City'          => $this->get_value_by_key( $record, 'venue_city' ),
-			'Country'       => $this->get_value_by_key( $record, 'venue_country' ),
-			'Province'      => $this->get_value_by_key( $record, 'venue_state' ),
-			'State'         => $this->get_value_by_key( $record, 'venue_state' ),
-			'Zip'           => $this->get_value_by_key( $record, 'venue_zip' ),
-			'Phone'         => $this->get_value_by_key( $record, 'venue_phone' ),
-			'URL'           => $this->get_value_by_key( $record, 'venue_url' ),
-			'ShowMap'       => $venue_id ? get_post_meta( $venue_id, '_VenueShowMap', true ) : $show_map_setting,
-			'ShowMapLink'   => $venue_id ? get_post_meta( $venue_id, '_VenueShowMapLink', true ) : $show_map_setting,
-			'FeaturedImage' => $this->get_featured_image( $venue_id, $record ),
+		$venue   = [];
+		$columns = [
+			'Venue'       => 'venue_name',
+			'Address'     => 'venue_address',
+			'City'        => 'venue_city',
+			'Country'     => 'venue_country',
+			'Description' => 'venue_description',
+			'Phone'       => 'venue_phone',
+			'Province'    => 'venue_state',
+			'State'       => 'venue_state',
+			'URL'         => 'venue_url',
+			'Zip'         => 'venue_zip',
+		];
+
+		foreach ( $columns as $name => $key ) {
+			// Don't overwrite unmapped columns.
+			if ( ! $this->has_value_by_key( $record, $key ) ) {
+				continue;
+			}
+
+			$value = '';
+
+			// Address is a concatenation of two possible values.
+			if ( 'venue_address' === $key ) {
+				$address_1 = trim( $this->get_value_by_key( $record, 'venue_address' ) );
+				$address_2 = trim( $this->get_value_by_key( $record, 'venue_address2' ) );
+				$value     = ( empty( $address_2 ) ) ? $address_1 : $address_1 . ' ' . $address_2;
+
+				continue;
+			} else {
+				$value = $this->get_value_by_key( $record, $key );
+			}
+
+			/**
+			 * Allows filtering of main values before setting.
+			 * Return boolean false to prevent importing that value.
+			 * @since TBD
+			 *
+			 * @param string $key    The key for the value we'll be importing.
+			 * @param string $value  The value we'll be importing.
+			 * @param array  $venue  The array of venue data we're modifying.
+			 * @param array  $record The event record from the import.
+			 */
+			$value = apply_filters(
```

Comment:
This is a nice addition since there might be a discrepancy between the record name and the import field name (what the import file calls it, what we call it in our code), would it be worth specifying, in the comments, what is what?